src/cmds/qstat.c

```diff
@@ -2223,7 +2322,56 @@ qstat -B [-f] [ server_name... ]\n";
 						any_failed = 1;
 						break;
 					}
+					if (E_opt == 1) {
+						/* Local Server */
+						if (server_out[0] == '\0' || (strcmp(server_out, def_server) == 0)) {
+							/* This is probably the first job id requested from primary server */
+							if (prev_server[0] == '\0')
+								strcpy(prev_server, def_server);
+							strncat(job_list, job_id_out, PBS_MAXCLTJOBID-1);
+							strncat(job_list, ",", 1);
+							if (optind != argc -1)
+								continue;
+							else {
+								free(query_job_list);
+								query_job_list = strdup(job_list);
+								job_list[0] = '\0';
+							}
+						}
+						else {
+							/* Remote server but jobs in continuation */
+							if ((prev_server[0] == '\0' ) || (strcmp(server_out, prev_server) == 0)) {
+								/* This is probably the first job id requested and not from primary server */
+								if (prev_server[0] == '\0')
+									strcpy(prev_server, server_out);
+								strncat(job_list, job_id_out, PBS_MAXCLTJOBID-1);
+								strncat(job_list, ",", 1);
+								if (optind != argc-1)
+									continue;
+								else {
+									/* It's a new remote server and the only job */
+									new_remote_server = 1;
+									free(query_job_list);
+									query_job_list = strdup(job_list);
+									job_list[0] = '\0';
+								}
+							}
+							/* A new remote server */
+							else {
+								new_remote_server = 1;
+								free(query_job_list);
+								query_job_list = strdup(job_list);
+								memset(job_list, 0, ((argc-1) * (PBS_MAXCLTJOBID+1)));
+								strncpy(job_list, job_id_out, PBS_MAXCLTJOBID-1);
+								strncat(job_list, ",", 1);
+							}
+						}
+					}
 				} else {  /* must be a destination-id */
+					if (E_opt == 1) {
```

Comment:
Hi Vinod, 
     This behaviour is not documented in the EDD yet. Changes Look fine to me
I sign-off, if this behaviour is going get written in the EDD.